crates/walrus-sui/src/client/retry_client/retriable_sui_client.rs

```diff
@@ -305,79 +305,49 @@ impl RetriableSuiClient {
                 .filter(|coin: &Coin| future::ready(!exclude.contains(&coin.coin_object_id)))
         );
 
-        let mut selected_coins = Vec::with_capacity(max_num_coins);
+        // Limit the initial capacity to at most `MAX_GAS_PAYMENT_OBJECTS`, so that we don't try
+        // to allocate too much space if e.g. `usize::MAX` is passed as `max_num_coins`.
+        let mut selected_coins_heap: BinaryHeap<Reverse<OrderedCoin>> =
+            BinaryHeap::with_capacity(max_num_coins.min(MAX_GAS_PAYMENT_OBJECTS));
```

Comment:
Suggestion: Don't set a capacity at all as we probably only need a single-digit number of coins in the vast majority of cases.

Reply:
Also, nit: Given we now *only* have the heap, we can just call it `selected_coins`.